### ISSUE TITLE
Reduce warnings in unit tests

### ISSUE THREAD

**Issue by not522** [optuna/optuna#3815]
### Motivation

The Optuna's unit tests raise many warnings, which make it hard to recognize unexpected warning messages.

### Description

We can filter expected warnings using `@pytest.mark.filterwarnings` or `warnings.simplefilter("ignore")`. Some warnings may be unexpected or unsuitable. We should fix tests or Optuna's codes in such cases.

This issue is contribution-welcome. We welcome any fixes, even how trivial.

**Comment by jmsykes83** [comment]
I will work on this.

**Comment by nzw0301** [comment]
Hi @jmsykes83 thank you for letting us know your interest! I'm also working on this prioritised issue too since this issue affects many developers' experiences. Do you have specific modules you would like to resolve this issue? I'd like to avoid conflicts with your forthcoming pull request.

**Comment by jmsykes83** [comment]
I will let you take the lead on this one.  Was there any direction or ideas you had in approaching this.

**Comment by nzw0301** [comment]
Thank you for your reply. How about some tests in `tests/visualization_tests`, maybe `tests/visualization_tests/test_utils.py` is the easiest one? The expected change will be similar to changes introduced by the linked PRs with this PR such as adding `@pytest.mark.filterwarnings` decorator to a test file.

**Comment by jmsykes83** [comment]
I know I can do this:
`
@pytest.mark.filterwarnings("ignore:Trial0")
def test_filter_nonfinite_with_invalid_target() -> None:
    study = prepare_study_with_trials()
    trials = study.get_trials(states=(TrialState.COMPLETE,))
    with pytest.raises(ValueError):
        _filter_nonfinite(trials, target=lambda t: "invalid target")  # type: ignore
`
But is this the solution you're looking for.  This seems like a solution that could become very cluttered in the long run.  This filters the only warning I am getting for pytest tests/visualization_tests/test_utils.py

**Comment by nzw0301** [comment]
In my understanding, the expected change is something like this. If we use stronger `filterwarnings`, we might miss unintended warning messages, so such a minimal filter looks enough.

**Comment by jmsykes83** [comment]
Did we need to filter the `user warnings` for `test_parallel_coordinate.py` as well?  If so, all of them or just the ones with the `  Attempting to set identical bottom == top == 29.999999999999996 results in singular transformations; automatically expanding?`

**Comment by nzw0301** [comment]
I'd suggest sending a PR modifying at most a few files such as only `tests/visualization_tests/test_utils.py` at first, not for all visualsation tests.

**Comment by jmsykes83** [comment]
> I'd suggest sending a PR modifying at most a few files such as only `tests/visualization_tests/test_utils.py` at first, not for all visualsation tests.

This is what I am doing

**Comment by jmsykes83** [comment]
So we don't work on the same test.  I will be sending in a pr once I finish some final tests.
These are the files I worked on:
```
        modified:   optuna/visualization/matplotlib/_parallel_coordinate.py
        modified:   tests/visualization_tests/matplotlib_tests/test_parallel_coordinate.py
        modified:   tests/visualization_tests/matplotlib_tests/test_pareto_front.py
        modified:   tests/visualization_tests/test_contour.py
        modified:   tests/visualization_tests/test_utils.py
```

**Comment by nzw0301** [comment]
Some tests will be updated mainly by the ongoing PRs: https://github.com/optuna/optuna/pulls?q=is%3Apr+is%3Aopen+label%3Av3. So you might not need to work on them. (This is why I said 

> How about some tests in tests/visualization_tests, maybe tests/visualization_tests/test_utils.py is the easiest one?

**Comment by nzw0301** [comment]
Except for the ongoing PRs linked above, the following files/directories should be addressed to resolve this issue with my (almost meaningless) comments. 

I don't leave comments either reason: i) I could not run several integration tests due to my local laptop issue and ii) the visualization module tests will be largely refactored by the forthcoming v3-related PRs.

```bash
├── integration_tests
│   ├── allennlp_tests 
│   │   └── test_allennlp.py # many deprecated warning messages from torchvision?
│   ├── lightgbm_tuner_tests # deprecated callback 
│   ├── test_catalyst.py
│   ├── test_chainermn.py
│   ├── test_keras.py
│   ├── test_lightgbm.py  # deprecated callback 
│   ├── test_mxnet.py
│   ├── test_pytorch_distributed.py 
│   ├── test_pytorch_lightning.py
│   ├── test_tensorboard.py
│   ├── test_tensorflow.py
│   └── test_tfkeras.py
├── samplers_tests
│   ├── test_samplers.py # botorch and TPE-related warning
│   └── tpe_tests # TPE-related warning message.
└── visualization_tests
    ├── __init__.py
    ├── matplotlib_tests
    │   ├── __init__.py
    │   ├── test_contour.py
    │   ├── test_optimization_history.py
    │   ├── test_parallel_coordinate.py
    │   ├── test_param_importances.py
    │   ├── test_pareto_front.py
    │   └── test_slice.py
    ├── test_contour.py
    ├── test_edf.py
    ├── test_intermediate_plot.py
    ├── test_optimization_history.py
    ├── test_parallel_coordinate.py
    ├── test_param_importances.py
    ├── test_pareto_front.py
    └── test_slice.py
```

**Comment by jmsykes83** [comment]
Ok, If I understand correctly.  You want only the ones with your comments on them addressed?  Is this correct? Or this is just a compilation of the files that need to be addressed.  I wonder this, because I see some not test files like _init_.py.  In the compilation.

**Comment by nzw0301** [comment]
The listings above are based on the output of `tree` command without any additional filtering, so some files do not show any warning message.

**Comment by jmsykes83** [comment]
Can you give me a list of files you want me to work on?  This way we are in sync.  I don't want to have overlapping effort.

**Comment by nzw0301** [comment]
how about files in `visualization_tests`?

**Comment by jmsykes83** [comment]
This is fine.  I will work on these.

**Comment by not522** [comment]
Since many simple warnings have been resolved, let me remove the `good first issue` label.

**Comment by sousu4** [comment]
I executed the following command:
```
pytest tests -m "not integration & not slow"
```

As a result, there were five RuntimeWarnings, two UserWarnings, and one InputDataWarning outputted. Here are the details:

```
tests/samplers_tests/test_nsgaiii.py::test_crossover_objectives[2-crossover2-2]
tests/samplers_tests/test_nsgaiii.py::test_crossover_objectives[2-crossover4-2]
tests/samplers_tests/test_nsgaiii.py::test_crossover_objectives[3-crossover4-2]
tests/samplers_tests/test_crossover_dims[2-crossover1-2]
tests/samplers_tests/test_crossover_dims[2-crossover2-2]
tests/samplers_tests/test_crossover_dims[2-crossover3-2]
tests/samplers_tests/test_crossover_dims[3-crossover4-2]
  /Users/minamiyoshiki/Documents/optuna/venv/lib/python3.8/site-packages/optuna/samplers/_nsgaiii.py:423: RuntimeWarning:
  
  divide by zero encountered in divide

tests/samplers_tests/tpe_tests/test_multi_objective_sampler.py::test_calculate_weights_below_for_multi_objective
  /Users/minamiyoshiki/Documents/optuna/venv/lib/python3.8/site-packages/optuna/_hypervolume/utils.py:14: RuntimeWarning:
  
  invalid value encountered in subtract

tests/samplers_tests/tpe_tests/test_truncnorm.py::test_ppf[-inf-inf]
tests/samplers_tests/tpe_tests/test_truncnorm.py::test_ppf[-10-10]
tests/samplers_tests/tpe_tests/test_truncnorm.py::test_ppf[-1-1]
tests/samplers_tests/tpe_tests/test_truncnorm.py::test_ppf[-0.001-0.001]
tests/samplers_tests/tpe_tests/test_truncnorm.py::test_ppf[-100--10]
  /Users/minamiyoshiki/Documents/optuna/venv/lib/python3.8/site-packages/optuna/samplers/_tpe/_truncnorm.py:181: RuntimeWarning:
  
  divide by zero encountered in log

tests/samplers_tests/tpe_tests/test_truncnorm.py::test_ppf[10-100]
tests/samplers_tests/tpe_tests/test_truncnorm.py::test_ppf[0-0]
  /Users/minamiyoshiki/Documents/optuna/venv/lib/python3.8/site-packages/optuna/samplers/_tpe/_truncnorm.py:185: RuntimeWarning:
  
  divide by zero encountered in log1p

tests/samplers_tests/tpe_tests/test_truncnorm.py::test_ppf[0-0]
tests/samplers_tests/tpe_tests/test_truncnorm.py::test_logpdf[0-0]
  /Users/minamiyoshiki/Documents/optuna/venv/lib/python3.8/site-packages/optuna/samplers/_tpe/_truncnorm.py:55: RuntimeWarning:
  
  divide by zero encountered in log1p

tests/storages_tests/test_heartbeat.py::test_retry_failed_trial_callback_repetitive_failure[sqlite]
tests/storages_tests/test_heartbeat.py::test_retry_failed_trial_callback_repetitive_failure[cached_sqlite]
  /Users/minamiyoshiki/Documents/optuna/study/study.py:530: UserWarning:
  
  Heartbeat of storage is supposed to be used with Study.optimize.

tests/study_tests/test_study.py::test_optimize_progbar_no_constraints[1]
tests/study_tests/test_study.py::test_optimize_progbar_no_constraints[2]
  /Users/minamiyoshiki/Documents/optuna/progress_bar.py:51: UserWarning:
  
  Progress bar won't be displayed because n_trials and timeout are None.

tests/terminator_tests/improvement_tests/gp_tests/test_botorch.py: 1 warning
tests/visualization_tests/test_terminator_improvement.py: 28 warnings
  /Users/minamiyoshiki/Documents/optuna/venv/lib/python3.8/site-packages/botorch/models/utils/assorted.py:201: InputDataWarning:
  
  Input data is not standardized. Please consider scaling the input to zero mean and unit variance.
```

**Comment by sousu4** [comment]
I'm planning to address the resolution of the two UserWarnings.
